zapread.com/Controllers/MessagesController.cs

```diff
@@ -873,18 +873,30 @@ public async Task<JsonResult> SendMessage(int id, string content, bool? isChat)
         }
 
         [HttpPost]
-        [AllowAnonymous]
         public JsonResult GetMessage(int id, int userId)
         {
+            var localUserId = User.Identity.GetUserId();
+
+            if(localUserId == null)
+            {
+                return Json(new { success = false, result = "Failure", message = "Error receiving message." });
```

Comment:
1.  Should change HTTP response code and have more informative message before returning.
`Response.StatusCode = (int)HttpStatusCode.Unauthorized;`
1.  Should have a descriptive message. `Error verifying logged in user.` or something?